tree/tree/inc/TTree.h

```diff
@@ -303,8 +309,8 @@ class TTree : public TNamed, public TAttLine, public TAttFill, public TAttMarker
    virtual TFriendElement *AddFriend(const char* treename, const char* filename = "");
    virtual TFriendElement *AddFriend(const char* treename, TFile* file);
    virtual TFriendElement *AddFriend(TTree* tree, const char* alias = "", Bool_t warn = kFALSE);
-   virtual void            AddTotBytes(Int_t tot) { fTotBytes += tot; }
-   virtual void            AddZipBytes(Int_t zip) { fZipBytes += zip; }
+   virtual void            AddTotBytes(Int_t tot) { std::lock_guard<std::mutex> sentry(fCounterMutex); fTotBytes += tot; }
```

Comment:
For production code, we would need to make those optional as we need to not penalize the serial code or even multi-thread code not using the IMT.